extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -441,7 +442,17 @@ export class AmpA4A extends AMP.BaseElement {
           // on precisely the same creative that was validated
           // via #validateAdResponse_.  See GitHub issue
           // https://github.com/ampproject/amphtml/issues/4187
-          return creativeDecoded && this.getAmpAdMetadata_(creativeDecoded);
+          let creativeMetaDataDef;
+          if (creativeDecoded &&
+            (creativeMetaDataDef = this.getAmpAdMetadata_(creativeDecoded))) {
+            // Load any extensions, do not wait on their promises as this
+            // is just to prefetch.
+            const extensions = extensionsFor(this.win);
+            creativeMetaDataDef.customElementExtensions.forEach(
+              extensionId => extensions.loadExtension(extensionId));
+            return creativeMetaDataDef;
+          }
+          return null;
```

Comment:
imho, it'd be clearer to return `null` up top if `creativeDecoded` or `getAmpMetadata` are falsy.